module/zcommon/zfs_fletcher.c

```diff
@@ -404,7 +386,7 @@ fletcher_4_native_impl(const fletcher_4_ops_t *ops, const void *buf,
 		ops->fini_native(zcp);
 }
 
-void
+inline void
 fletcher_4_native(const void *buf, uint64_t size, zio_cksum_t *zcp)
 {
```

Comment:
Please change line 407 to scalar, this would cause unnecessary recursion. The same goes the byteswap version.